src/lazy/expanded/macro_evaluator.rs

```diff
@@ -303,113 +450,46 @@ impl<
     /// Attempts to resolve the provided `invocation` in the specified `context`. Upon success,
     /// returns an iterator that lazily computes the expansion of the macro invocation and yields
     /// its values.
-    pub(crate) fn evaluate<'iter, 'top>(
+    pub fn evaluate<'iter>(
         &'iter mut self,
         context: EncodingContext<'top>,
-        invocation: M,
-    ) -> IonResult<EvaluatingIterator<'iter, 'top, 'data, D, M, S>> {
-        self.push(context, invocation)?;
+        invocation: impl Into<MacroExpr<'top, 'data, D>>,
+    ) -> IonResult<EvaluatingIterator<'iter, 'top, 'data, D>>
+    where
+        'data: 'top,
+        Self: Sized,
+    {
+        self.push(context, invocation.into())?;
         Ok(EvaluatingIterator::new(self, context))
     }
 }
 
-// ===== Type aliases for commonly used flavors of `MacroEvaluator` =====
```

Comment:
🗺️ `MacroEvaluator` is now a `struct` instead of a `trait`, so we no longer need these implementations.